include/range/v3/action/slice.hpp

```diff
@@ -36,47 +36,100 @@ namespace ranges
                     ranges::action::erase(rng, begin(rng), next(begin(rng), from));
-                    return static_cast<Rng&&>(rng);
+                    return static_cast<Rng &&>(rng);
                 }
 
+              template<typename Rng,
+                  typename I = iterator_t<Rng>,
+                  typename D = range_difference_type_t<Rng>,
+                  CONCEPT_REQUIRES_(BidirectionalRange<Rng>() && ErasableRange<Rng, I, I>())>
+              Rng operator()(Rng &&rng, D from, detail::from_end_<D> to) const {
+                  RANGES_EXPECT(0 <= from && to.dist_ <= 0);
+                  ranges::action::erase(rng, begin(rng), next(begin(rng), from));
+                  ranges::action::erase(rng, prev(end(rng), -to.dist_), end(rng));
+                  return static_cast<Rng &&>(rng);
+              }
+
+              template<typename Rng,
+                  typename I = iterator_t<Rng>,
+                  typename D = range_difference_type_t<Rng>,
+                  CONCEPT_REQUIRES_(BidirectionalRange<Rng>() && ErasableRange<Rng, I, I>())>
+              Rng operator()(Rng &&rng, detail::from_end_<D> from, detail::from_end_<D> to) const {
+                  RANGES_EXPECT(from.dist_ <= 0 && to.dist_ <= 0 && from.dist_ <= to.dist_);
+                  ranges::action::erase(rng, begin(rng), prev(end(rng), -from.dist_));
+                  ranges::action::erase(rng, prev(end(rng), -to.dist_), end(rng));
+                  return static_cast<Rng &&>(rng);
+              }
+
+              template<typename Rng,
+                  typename I = iterator_t<Rng>,
+                  typename D = range_difference_type_t<Rng>,
+                  CONCEPT_REQUIRES_(ForwardRange<Rng>() && ErasableRange<Rng, I, I>())>
+              Rng operator()(Rng &&rng, D from, const end_fn &) const {
+                  RANGES_EXPECT(0 <= from);
+                  ranges::action::erase(rng, begin(rng), next(begin(rng), from));
+                  return static_cast<Rng &&>(rng);
+              }
+
+              template<typename Rng,
+                  typename I = iterator_t<Rng>,
+                  typename D = range_difference_type_t<Rng>,
+                  CONCEPT_REQUIRES_(BidirectionalRange<Rng>() && ErasableRange<Rng, I, I>())>
+              Rng operator()(Rng &&rng, detail::from_end_<D> from, const end_fn &) const {
+                  RANGES_EXPECT(from.dist_ <= 0);
+                  ranges::action::erase(rng, begin(rng), prev(end(rng), -from.dist_));
+                  return static_cast<Rng &&>(rng);
+              }
+
+              template<typename A>
+              using is_end_type = meta::or_<meta::is<A, detail::from_end_>, std::is_same<A, end_fn>>;
+
             #ifndef RANGES_DOXYGEN_INVOKED
                 template<typename Rng, typename T, typename U,
-                    CONCEPT_REQUIRES_(!Concept<Rng, T, U>())>
+                    CONCEPT_REQUIRES_(!ForwardRange<Rng>())>
```

Comment:
This isn't constrained to match exactly the cases that the other overloads do not admit.